compiler/rustc_lexer/src/lib.rs

```diff
@@ -388,6 +389,10 @@ impl Cursor<'_> {
                     let kind = RawByteStr { n_hashes: res.ok() };
                     Literal { kind, suffix_start }
                 }
+                ('u', 'i') if self.remaining_str().starts_with("uiltin#") => {
```

Comment:
I'm not adding any new token kinds here: I'm just making sure no `UnknownPrefix` token is emitted but instead the tokens are emitted piece by piece, like they used to be before they were marked as reserved in [RFC 3101](https://rust-lang.github.io/rfcs/3101-reserved_prefixes.html) / PR #85359. I've thought that was the point of the RFC: enabling custom syntaxes on top of it. In the whole discussion around `builtin#` it seemed to me that this RFC was the point of using the `#` instead of some other character, so it seems a bit weird to me to now use `builtin #` instead.

If you prefer, I can change the arm to `_ if self.remaining_str().starts_with("uiltin#")`, but I'm not sure of the performance impacts of that. Shrug.

Reply:
>[RFC 3101](https://rust-lang.github.io/rfcs/3101-reserved_prefixes.html) / PR https://github.com/rust-lang/rust/pull/85359 ... I've thought that was the point of the RFC: enabling custom syntaxes on top of it.

I'd say the goal now is to add *some* syntax with the main requirement of not causing troubles, not necessarily on top of RFC 3101.
Lexer hacks like this seem like a trouble.
But let me check what RFC 3101 is about.

Reply:
IDK from what I can tell the point of RFC 3101 was that `foo#` before could be used in macros. It was reserved, yes, mainly for custom literal types and such, but it wouldn't have been a stretch to use it for `builtin#` as well. The RFC mentions C and python reserving identifiers for language special use:

> The notion of reserving "syntactic space" as an aid to backwards-compatibility is an idea that has precedence from other languages. C reserves large swathes of the identifier space for its own use, most notably identifiers that begin with _ or __. Likewise, Python reserves all identifiers of the form `__foo__` for special use by the language.

With RFC 3101 one would have had `builtin#foo` in the territory of errors, at least in the 2021 edition and beyond. With `builtin # offset_of` that could have previously been used inside custom macro syntax, although unlikely. But shrug, I'll change it to `builtin # foo`.

In theory, thinking of it, a nicer implementation should probably have emitted a special `BuiltinPrefix` token or such, to only allow `builtin#foo` or `builtin# foo` syntax, but keep `builtin # foo` to macros. This PR would have still been a breaking change for code on the 2018/2015 editions, but that could have been easily avoided by an edition specific check.